[CI/Build] adopt uv

kemingy opened the pull request:
FIX https://github.com/vllm-project/production-stack/issues/184 

**BEFORE SUBMITTING, PLEASE READ THE CHECKLIST BELOW AND FILL IN THE DESCRIPTION ABOVE**

---

- [x] Make sure the code changes pass the [pre-commit](../CONTRIBUTING.md) checks.
- [x] Sign-off your commit by using <code>-s</code> when doing <code>git commit</code>
- [x] Try to classify PRs for easy understanding of the type of changes, such as `[Bugfix]`, `[Feat]`, and `[CI]`.

<details>
<!-- inside this <details> section, markdown rendering does not work, so we use raw html here. -->
<summary><b> Detailed Checklist (Click to Expand) </b></summary>

<p>Thank you for your contribution to production-stack! Before submitting the pull request, please ensure the PR meets the following criteria. This helps us maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Please try to classify PRs for easy understanding of the type of changes. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Feat]</code> for new features in the cluster (e.g., autoscaling, disaggregated prefill, etc.).</li>
    <li><code>[Router]</code> for changes to the <code>vllm_router</code> (e.g., routing algorithm, router observability, etc.).</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>Pass all linter checks. Please use <code>pre-commit</code> to format your code. See <code>README.md</code> for installation.</li>
    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li> Please include sufficient tests to ensure the change is stay correct and robust. This includes both unit tests and integration tests.</li>
</ul>

<h3>DCO and Signed-off-by</h3>
<p>When contributing changes to this project, you must agree to the <a href="https://github.com/vllm-project/vllm/blob/main/DCO">DCO</a>. Commits must include a <code>Signed-off-by:</code> header which certifies agreement with the terms of the DCO.</p>
<p>Using <code>-s</code> with <code>git commit</code> will automatically add this header.</p>

<h3>What to Expect for the Reviews</h3>

We aim to address all PRs in a timely manner. If no one reviews your PR within 5 days, please @-mention one of YuhanLiu11
, Shaoting-Feng or ApostaC.

YuhanLiu11 commented:
@kemingy Thanks for the contribution! We will be looking at this soon! 

cc @Shaoting-Feng

Shaoting-Feng reviewed (commented):
Thanks for your contribution! Could you help resolve the conflicts? Over the weekend, we added three packages for the router. Could you also include them in the `pyproject.toml` file?

kemingy commented:
> Thanks for your contribution! Could you help resolve the conflicts? Over the weekend, we added three packages for the router. Could you also include them in the `pyproject.toml` file?

Rebased. PTAL.

Shaoting-Feng commented:
Thanks for your fix. Could you help make the optioinal dependencies work consistently with the docker build phase? For example, the INSTALL_SENTENCE_TRANSFORMERS argument in the functionality test workflow.

kemingy commented:
> Thanks for your fix. Could you help make the optioinal dependencies work consistently with the docker build phase? For example, the INSTALL_SENTENCE_TRANSFORMERS argument in the functionality test workflow.

Fixed.

bufferoverflow commented on pyproject.toml:
```diff
@@ -1,3 +1,42 @@
+[project]
+name = "production-stack"
```
?, see https://github.com/vllm-project/production-stack/issues/178 , right now this is[vlllm-router](https://github.com/vllm-project/production-stack/blob/main/setup.py#L4)

kemingy replied:
fixed

pwuersch commented on CONTRIBUTING.md:
````diff
@@ -39,15 +39,15 @@ Follow the standard GitHub workflow:
 ```bash
-pip install -r requirements-lint.txt
-pre-commit install
+uv sync uv sync --all-extras --all-groups
````
```suggestion
uv sync --all-extras --all-groups
```

pwuersch commented on pyproject.toml:
```diff
@@ -1,3 +1,42 @@
+vllm-router = "vllm_router.app:main"
+
+[project.urls]
+Repository = "ttps://github.com/vllm-project/production-stack"
```
```suggestion
Repository = "https://github.com/vllm-project/production-stack"
```

pwuersch reviewed (changes requested):
I think there are two small oversights

Shaoting-Feng commented:
Thanks for your fix! Could you fix the DCO? Then we're good to go.

kemingy commented:
Hi @Shaoting-Feng, DCO has been fixed.

Shaoting-Feng reviewed (approved):
LGTM. Thanks!
